C#/Botkit/Microsoft.BotKit/BotWorker.cs

```diff
@@ -104,17 +103,41 @@ public async Task<object> Reply(IBotkitMessage message, string response)
         /// </summary>
         /// <param name="id">ID of dialog</param>
         /// <param name="options">Object containing options to be passed into the dialog</param>
-        public async void BeginDialog(string id, object options)
+        public async Task BeginDialogAsync(string id, Dictionary<string, object> options)
         {
+            if (config.DialogContext != null)
```

Comment:
Can we refactor this with the next format:
```
if (config.DialogContext != null)
{
    throw new Exception("Call to beginDialog on a bot that did not receive a 
    dialogContext during spawn");
}

var opt = new Dictionary<string, object>();
```

Reply:
Done